test/libsolutil/Whiskers.cpp

```diff
@@ -88,8 +88,13 @@ BOOST_AUTO_TEST_CASE(conditional_with_else)
 BOOST_AUTO_TEST_CASE(broken_conditional_with_else)
 {
 	string templ = "<?b>X<!bY</b>";
-	BOOST_CHECK_EQUAL(Whiskers(templ)("b", true).render(), "X<!bY");
-	BOOST_CHECK_EQUAL(Whiskers(templ)("b", false).render(), "");
+	BOOST_CHECK_THROW(Whiskers{templ}, WhiskersError);
+
+	templ = "<?bX<!b>Y</b>";
+	BOOST_CHECK_THROW(Whiskers{templ}, WhiskersError);
+
+	templ = "<?b>X<!b>Y</b";
+	BOOST_CHECK_THROW(Whiskers{templ}, WhiskersError);
```

Comment:
Would be nice if we could inspect the exception message, but didn't find such a feature in boost after a quick look.